src/noInputRenameRule.ts

```diff
@@ -34,7 +36,11 @@ export class InputMetadataWalker extends NgWalker {
     const className = (property.parent as ts.PropertyAccessExpression).name.getText();
     const memberName = property.name.getText();
 
-    if (args.length === 0 || (this.directiveSelector && this.directiveSelector.indexOf(memberName) !== -1)) {
+    if (
+      args.length === 0 ||
+      (this.directiveSelector &&
+        (input.expression as any).arguments.some(arg => this.directiveSelector.indexOf(arg.text) !== -1 && memberName !== arg.text))
```

Comment:
Instead of casting to `any`, what about `(input.expression as ts.CallExpression)`?